packages/hebao_v1/contracts/stores/SecurityStore.sol

```diff
@@ -53,46 +55,93 @@ contract SecurityStore is DataStore
         view
         returns (bool)
     {
-        return wallets[wallet].guardianIdx[guardian] > 0;
+        return wallets[wallet].guardianIdx[guardian] > 0 &&
+            wallets[wallet].guardianEffectiveTime[guardian] >= now &&
+            wallets[wallet].guardianRemovalEffectiveTime[guardian] < now;
     }
 
     function getGuardian(
         address wallet,
-        address _guardian
+        address guardian
         )
         public
         view
-        returns (Data.Guardian memory guardian)
+        returns (Data.Guardian memory)
     {
-        uint index = wallets[wallet].guardianIdx[_guardian];
+        Wallet storage w = wallets[wallet];
+        uint index = w.guardianIdx[guardian];
         require(index > 0, "NOT_A_GUARDIAN");
-        guardian = wallets[wallet].guardians[index-1];
+        require(
+            w.guardianEffectiveTime[guardian] >= now &&
+            w.guardianRemovalEffectiveTime[guardian] < now,
+            "NOT_A_EFFECTIVE_GUARDIAN"
+        );
+        return wallets[wallet].guardians[index-1];
     }
 
     function guardians(address wallet)
         public
         view
-        returns (Data.Guardian[] memory)
+        returns (Data.Guardian[] memory effectiveGuardians)
     {
-        return wallets[wallet].guardians;
+        Wallet storage w = wallets[wallet];
+        effectiveGuardians = new Data.Guardian[](w.guardians.length);
+        uint index = 0;
+        for (uint i = 0; i < w.guardians.length; i++) {
+            Data.Guardian memory g = w.guardians[i];
+            if (w.guardianEffectiveTime[g.addr] >= now &&
+                w.guardianRemovalEffectiveTime[g.addr] < now) {
+                effectiveGuardians[index] = g;
+                index ++;
+            }
+        }
     }
 
     function numGuardians(address wallet)
         public
         view
         returns (uint)
     {
-        return wallets[wallet].guardians.length;
+        Wallet storage w = wallets[wallet];
+        uint num = 0;
+        for (uint i = 0; i < w.guardians.length; i++) {
+            Data.Guardian memory g = w.guardians[i];
+            if (w.guardianEffectiveTime[g.addr] >= now &&
+                w.guardianRemovalEffectiveTime[g.addr] < now) {
+                num ++;
+            }
+        }
+        return num;
```

Comment:
Maybe this can just do `return guardians(wallet).length;` which won't be significantly more expensive.